corehq/apps/app_manager/tests/app_factory.py

```diff
@@ -81,20 +99,25 @@ def form_updates_case(form, case_type=None, parent_case_type=None, update=None):
                 )
                 module.parent_select.module_id = parent_select_module.unique_id
         else:
-            case_type = case_type or form.get_module().case_type
-            index = len([load for load in form.actions.load_update_cases if load.case_type == case_type])
-            action = LoadUpdateAction(
-                case_type=case_type,
-                case_tag='load_{}_{}'.format(case_type, index),
-                case_properties=update,
-            )
+            AppFactory._append_loadupdate_action(form, case_type, parent_case_type, update=update)
 
-            if parent_case_type:
-                parent_action = form.actions.load_update_cases[-1]
-                assert parent_action.case_type == parent_case_type
-                action.case_index = CaseIndex(tag=parent_action.case_tag)
+    @staticmethod
+    def form_preloads_case(form, case_type=None, parent_case_type=None, preload=None):
+        if form.form_type == 'module_form':
+            form.requires = 'case'
+            form.actions.case_preload = PreloadAction(preload=preload)
+            form.actions.case_preload.condition.type = 'always'
 
-            form.actions.load_update_cases.append(action)
+            if parent_case_type:
+                module = form.get_module()
+                module.parent_select.active = True
+                parent_select_module = next(
+                    module for module in module.get_app().get_modules()
+                    if module.case_type == parent_case_type
+                )
+                module.parent_select.module_id = parent_select_module.unique_id
+        else:
+            return AppFactory._append_loadupdate_action(form, case_type, parent_case_type, preload=preload)
```

Comment:
Sorry to be nitpicky but doing it this way will create two actions if the user wants to do preloading and updating for advanced forms:
```
factory.form_preloads_case(...)
factory.form_updates_case(...)
```

Perhaps the simpler approach is to just add a `preload` arg to the the `form_updates_case` function.